tika-parsers/tika-parsers-classic/tika-parsers-classic-modules/tika-parser-zip-commons/src/main/java/org/apache/tika/detect/zip/DefaultZipContainerDetector.java

```diff
@@ -224,29 +230,53 @@ MediaType detectStreaming(InputStream input, Metadata metadata) throws IOExcepti
         } catch (Exception e) {
             //swallow
         }
-
+        
         return finalDetect(detectContext);
     }
-
-
+    
+    
     private MediaType detect(ZipArchiveEntry zae, ZipArchiveInputStream zis,
-                             StreamingDetectContext detectContext) throws IOException {
-        for (ZipContainerDetector d : zipDetectors) {
+            StreamingDetectContext detectContext) throws IOException {
+        for (ZipContainerDetector d : getDetectors()) {
             MediaType mt = d.streamingDetectUpdate(zae, zis, detectContext);
             if (mt != null) {
                 return mt;
             }
         }
         return null;
     }
-
+    
     private MediaType finalDetect(StreamingDetectContext detectContext) {
-        for (ZipContainerDetector d : zipDetectors) {
+        for (ZipContainerDetector d : getDetectors()) {
             MediaType mt = d.streamingDetectFinal(detectContext);
             if (mt != null) {
                 return mt;
             }
         }
         return MediaType.APPLICATION_ZIP;
     }
+    
+    private void setDetectors(List<ZipContainerDetector> zipDetectors) {
```

Comment:
There's a functional mismatch between setDetectors and getDetectors.

Best is, to remove this method and to let the constructor access the field directly.